src/client/sandbox/event/message.js

```diff
@@ -33,9 +33,12 @@ export default class MessageSandbox extends SandboxBase {
         this.topWindow = null;
         this.window    = null;
 
-        this.listeners = listeners;
+        this.listeners     = listeners;
+        this.unloadSandbox = unloadSandbox;
 
         this.storedOnMessageHandler = null;
+
+        this.serviceMsgsQueue = [];
```

Comment:
Is this client-server service messages or cross-domain communication service messages?
If first, we should move this code to `transport`, if second we should rename them somehow to distinct them from client-server XHR methods.